src/charm.py

```diff
@@ -0,0 +1,216 @@
+
+        Args:
+            event:The event triggered by the restart action
+        """
+        container = self.unit.get_container(self.name)
+        if not container.can_connect():
+            event.defer()
+            return
+
+        self.unit.status = MaintenanceStatus("restarting openldap")
+        container.restart(self.name)
+        event.set_results({"result": "openldap successfully restarted"})
+        self.unit.status = ActiveStatus()
+
+    def _on_get_admin_password(self, event):
+        """Get admin password, action handler.
+
+        Args:
+            event:The event triggered by the `get-admin-password` action.
+        """
+        admin_password = self._state.bind_password
+        event.set_results({"admin-password": admin_password})
+
+    def validate(self):
+        """Validate that configuration and relations are valid and ready.
+
+        Raises:
+            ValueError: in case of invalid configuration.
+        """
+        if not self._state.is_ready():
+            raise ValueError("peer relation not ready")
+
+    def update(self, event):
+        """Update the openldap server configuration and re-plan its execution.
+
+        Args:
+            event: The event triggered when the relation changed.
+        """
+        try:
+            self.validate()
+        except ValueError as err:
+            self.unit.status = BlockedStatus(str(err))
+            return
+
+        container = self.unit.get_container(self.name)
+        if not container.can_connect():
+            event.defer()
+            return
+
+        container.push_path("templates/", "/")
+        self.model.unit.open_port(port=APPLICATION_PORT, protocol="tcp")
+
+        logger.info("configuring openldap")
+
+        context = {}
+        for key, value in self.config.items():
+            updated_key = key.upper().replace("-", "_")
+            context[updated_key] = value
+
+        # Set provider values in state
+        self._state.bind_password = self._state.bind_password or random_string(
+            12
+        )
+        self._state.base_dn = self.config["ldap-base-dn"]
+
+        context.update(
+            {
+                "LDAP_ADMIN_PASSWORD": self._state.bind_password,
+                "LDAP_ADMIN_USERNAME": "admin",
+                "LDAP_TLS": "false",
+            }
+        )
+
+        logger.info("planning openldap execution")
+        pebble_layer = {
+            "summary": "openldap layer",
+            "services": {
+                self.name: {
+                    "summary": "openldap",
+                    "command": "/container/tool/run",
+                    "startup": "enabled",
+                    "override": "replace",
+                    "environment": context,
+                }
+            },
+        }
+        container.add_layer(self.name, pebble_layer, combine=True)
+        container.replan()
+
+        self.unit.status = ActiveStatus("Status check: UP")
```

Comment:
Perhaps remove `"Status check: UP"` since you don't have status checks implemented?